### PR TITLE
feat: supported HRMP transfer between Shiden and Crust shadow

### PR DESCRIPTION
**Pull request by yashirooooo**
**Pull Request Summary**

> Describe what changes this pull request makes to the repository

Pull Request Summary

- feat: supported HRMP transfer between Shiden and Crust shadow
  - feat: added CSM in Shiden
  - feat: supported CSM transfer between Shiden and Crust shadow

Progresses

- [x] listing HRMP assets on the native and EVM assets panel
- [x] depositing assets from Crust shadow to Shiden(EVM)
- [x] withdrawing assets from Shiden(Native) to Acala

**Check list**
- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**
image
image
image

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,151 +1,152 @@
-const { resolve } = require('path');
```
Comment (roy1210) [inline .eslintrc.js:1]: Why did you change this file?

Comment (impelcrypto) [inline .eslintrc.js:1]: Why did you change this file?

File: src/modules/xcm/tokens/index.ts [hunk 2]
```diff
@@ -177,6 +177,16 @@ export const xcmToken = {
...
     },
+    {
+      symbol: 'CSM',
+      isNativeToken: true,
+      assetId: '18446744073709551624',
+      originAssetId: 'CSM',
+      logo: require('/src/assets/img/shadow.svg'),
+      isXcmCompatible: true,
+      originChain: Chain.CRUST_SHADOW,
+      minBridgeAmount: '0.001',
```
Comment (fiexer) [inline src/modules/xcm/tokens/index.ts:188]: The minimum bridged amount is calculated to be 118.547 CSM based on our `unitsPerSecond`

File: src/v2/models/XcmModels.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ export enum Chain {
   STATEMINT = 'Statemint',
   KINTSUGI = 'Kintsugi',
   INTERLAY = 'Interlay',
+  CRUST_SHADOW = 'Crust-shadow',
```
Comment (fiexer) [inline src/v2/models/XcmModels.ts:17]: Can we do Crust Shadow?

File: src/v2/repositories/implementations/xcm/CrustShadowXcmRepository.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        },
+      },
+    };
+
+    //Memo: each XCM instruction is weighted to be 1_000_000_000 units of weight and for this op to execute
+    //weight value of 5 * 10^9 is generally good
+    const destWeight = new BN(10).pow(new BN(9)).muln(5);
+    return await this.buildTxCall(
+      from,
+      'xTokens',
```
Comment (gluneau) [inline src/v2/repositories/implementations/xcm/CrustShadowXcmRepository.ts:74]: Hello @yashirooooo We think `XTokens` is wrong, it seems you actually use `PolkadotXcm`

look into `src/v2/repositories/implementations/xcm/AstarXcmRepository.ts` for inspiration.

**Review by gluneau — APPROVED** [review]
I've tested many transfers in and out, using xcm, native & evm with great success.

**Review by sirius651 — APPROVED** [review]
I've also tested for whole cycle, works well on my end.

**Comment by sirius651** [comment]
Hi @yashirooooo could you resolve conflicts with `main`? once it done, it seems fine to merged

**Comment by yashirooooo** [comment]
> Hi @yashirooooo could you resolve conflicts with `main`? once it done, it seems fine to merged

I have solved it, you can take a look, thank you

**Comment by impelcrypto** [comment]
@yashirooooo @gluneau @sirius651 
I got this error when I try to deposit SDN from Crust to Shiden.

Screenshot 2022-11-28 at 2 03 59 PM

image

**Comment by yashirooooo** [comment]
> @yashirooooo @gluneau @sirius651 I got this error when I try to deposit SDN from Crust to Shiden.
> 
> Screenshot 2022-11-28 at 2 03 59 PM
> 
> image

I have fixed it. Please check it out in this pr #606

File: src/v2/models/XcmModels.ts (hunk 3 above)
Comment (impelcrypto) [reply src/v2/models/XcmModels.ts:17]: I've resolved it (for UI) in this PR.
https://github.com/AstarNetwork/astar-apps/pull/607

> fix: cast the network name from 'Crust-shadow' to 'Crust Shadow' (for UI)

Comment (impelcrypto) [reply src/v2/models/XcmModels.ts:17]: CC: @fiexer
